static/js/starred_messages.js

```diff
@@ -34,6 +35,17 @@ exports.get_starred_msg_ids = function () {
     return Array.from(exports.ids);
 };
 
+exports.unstar_topic = function (stream_id, topic) {
+    const all_starred = exports.get_starred_msg_ids();
+    all_starred.forEach((msg_id) => {
+        const message = message_store.get(msg_id);
+        if (message.stream_id === stream_id &&
+                message.topic.toLowerCase() === topic.toLowerCase()) {
+            message_flags.toggle_starred_and_update_server(message);
+        }
+    });
+};
```

Comment:
Can you look at https://github.com/zulip/zulip/pull/14753/files?  We should share code between the two, but I'm not sure which version is better.  Might be worth your making a PR with your revisions to #14753 to clean this bit out.